### PR TITLE
hack/update-translations.sh: Improve backslash handling

### LINKED ISSUE #1326: `hack/update-translations.sh -x` fails due to backslash in text
**What happened**:

`hack/update-translations.sh -x` is needed to update `template.pot` so that kubectl's strings can be translated to other languages. However it currently fails with "invalid control sequence" errors:

```
$ hack/update-translations.sh -x
Extracting strings to POT
tmp.pot:2221:67: invalid control sequence
tmp.pot:2221:73: invalid control sequence
tmp.pot:2222:58: invalid control sequence
tmp.pot:2222:64: invalid control sequence
msgcat: found 4 fatal errors
Failed to update template.pot
```

Investigating the cause, I've determined it is failing on these two lines:
https://github.com/kubernetes/kubernetes/blob/d72926d4915133550e41de1252fe2a4aa0aa3745/staging/src/k8s.io/kubectl/pkg/cmd/completion/completion.go#L105-L106
```
		    kubectl completion powershell > $HOME\.kube\completion.ps1
		    Add-Content $PROFILE "$HOME\.kube\completion.ps1"
```

~I believe Powershell can interpret paths that use `/` (need to confirm this?), so can we simply change the example from using `\` to `/` to work around this problem?~  Edit: I think I found a way to improve `\` handling to work as-is. See PR

**What you expected to happen**:
`hack/update-translations.sh -x` should run without an error

**How to reproduce it (as minimally and precisely as possible)**:
Run `hack/update-translations.sh -x` from the k/k repository root.

**Anything else we need to know?**:

**Environment**:
- Kubernetes client and server versions (use `kubectl version`): 1.26
- Cloud provider or hardware configuration: n/a
- OS (e.g: `cat /etc/os-release`): Ubuntu 22.04.1 LTS

### PR DESCRIPTION
**Pull request by brianpursley**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Replace specific backslash escape handling with generalized backslash handling.

Fixes bug where `\` would not be escaped when it should, resulting in `hack/update-translations.sh` failing or producing the wrong output.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1326

#### Special notes for your reviewer:
I regenerated `template.pot` now that the script works again.  This picked up a bunch of changes since it hasn't been updated in 17 months.

This doesn't update translations, but the updated `template.pot` enables translations to be updated again as it contains the updated msgid keys.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by brianpursley** [comment]
/assign @deads2k

Let me know if you think I should remove the `template.pot` changes and do that in a separate PR.  I mostly wanted to show that it is working again after making this change.

File: hack/update-translations.sh [hunk 1]
```diff
@@ -83,8 +83,7 @@ if [[ "${generate_pot}" == "true" ]]; then
   # shellcheck disable=SC2046
   go-xgettext -k=i18n.T $(grep -lr "i18n.T" "${KUBECTL_FILES[@]}" | grep -vE "${KUBECTL_IGNORE_FILES_REGEX}") > tmp.pot
   perl -pi -e 's/CHARSET/UTF-8/' tmp.pot
-  perl -pi -e 's/\\\(/\\\\\(/g' tmp.pot
-  perl -pi -e 's/\\\)/\\\\\)/g' tmp.pot
+  perl -pi -e 's/\\(?!n"\n|")/\\\\/g' tmp.pot
```
Comment (atiratree) [inline hack/update-translations.sh:86]: this seems reasonable

To summarize: we are trying to preserve all of our `\` symbols that might be used in for example escapes for shell commands and we want  to print them to the user. To preserve them we have to escape them with `\`. But xgettext puts its own `\` symbols around `"` and converts newlines into `\n` so we have to skip these two cases.

**Comment by brianpursley** [comment]
/unhold

I don't think I need to hold this since I'm not an approver in the hack directory.

**Comment by brianpursley** [comment]
@soltysh thoughts on this change?  It looks like you are an approver in `hack` and this falls under sig-cli's umbrella.

I think we either need to make this change, or remove the `\` from kubectl's examples (and prevent `\` from being used inside strings in the future).

I feel like this more generalized regex should work and allow us not to have to worry about `\` in the future.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**